### PR TITLE
Add debug feature to always enable cors for emulated v2 https functions

### PR DESCRIPTION
**Pull request by taeold**
As suggested by @jhuleatt, we add a debug feature in the SDK to enable CORS setting (i.e. set CORS origin setting to allow requests from any origin) during function emulation to make emulator experience less painful. For example, given the following v2 https function:

```js
const f = onRequest({ cors: ['mysite.com']})...
```

Users would've had to add `localhost:XXX` to the cors setting to make function emulation work. With the new debug feature (which the Functions emulator will enable), users won't need to make such change.

Thanks @jhuleatt for pushing the idea for an excellent feature!

### CODE REVIEW

**Review by colerogers — APPROVED** [review]

File: src/v2/providers/https.ts [hunk 1]
```diff
@@ -89,12 +90,13 @@ export function onRequest(
     opts = optsOrHandler as HttpsOptions;
   }
 
-  if ('cors' in opts) {
+  if (isDebugFeatureEnabled('enableCors') || 'cors' in opts) {
```
Comment (colerogers) [inline src/v2/providers/https.ts:93]: should we add a simple test case to this?

Comment (jhuleatt) [reply src/v2/providers/https.ts:93]: +1 to make sure we don't accidentally open someone's function up to the world

**Review by jhuleatt — APPROVED** [review]

**Review by inlined — APPROVED** [review]
I think we _also_ need to default cors to true to keep parity with V1.

**Comment by inlined** [comment]
Whoops. I see where we did & still do default to CORS true for Callable functions in v2.

**Pull request merged**